sympy/angles/euler.py

```diff
@@ -0,0 +1,688 @@
+        >>> euler = Euler(pi / 4, 0, 0, 'ZYX')
+        >>> euler.mul(2)
+        Euler(pi/2,  0,  0, ZYX)
+
+        Wrapping always happens, and the angles are between [-pi and pi],
+        so the result might seem unexpected:
+
+        >>> euler.mul(4)
+        Euler((-pi),  0,  0, ZYX)
+
+        Possible to compose Euler instances. For simple cases it might work
+        well:
+        >>> euler = Euler((-pi),  0,  0, 'ZYX')
+        >>> euler.mul(euler)
+        Euler(0,  0,  0, ZYX)
+
+        Most of the time, it is not very efficient:
+
+        >>> euler = Euler(pi / 4, 0, 0, 'ZYX')
+        >>> euler.mul(euler)
+        Euler(2*atan(2*sqrt(2)*sqrt(1/2 - sqrt(2)/4)*sqrt(sqrt(2)/4 + 1/2)),  0,  0, ZYX)
+
+        By evaluating it, we can see it is equivalent to the expected answer:
+
+        >>> euler.mul(euler).evalf()
+        Euler(1.5707963267949,  0,  0, ZYX)
+        """
+        return self._generic_mul(self, _sympify(other))
+
+    def __add__(self, other):
+        return self.add(other)
+
+    def __radd__(self, other):
+        return self.add(other)
+
+    def __sub__(self, other):
+        return self.add(-other)
+
+    def __mul__(self, other):
+        return self._generic_mul(self, _sympify(other))
+
+    def __rmul__(self, other):
+        return self._generic_mul(_sympify(other), self)
+
+    def __pow__(self, p):
+        return self.pow(p)
+
+    def __neg__(self):
+        return Euler(-self.alpha, -self.beta, -self.gamma, seq=self.seq)
+
+    def __truediv__(self, other):
+        return self * sympify(other)**-1
+
+    def __rtruediv__(self, other):
+        raise ValueError('Division by an Euler is not well defined.')
+
+    def inverse(self):
+        """Returns inverse Euler angles
+
+        In this implementation, the sequence is also reversed.
+        """
+        seq = self.seq[::-1]
+        alpha, beta, gamma = [-i for i in self.angles[::-1]]
+        return Euler(alpha, beta, gamma, seq)
+
+    def pow(self, p):
+        """Finds the pth power of the Euler angles.
+
+        Parameters
+        ==========
+
+        p : int
+            Power to be applied on Euler.
+
+        Returns
+        =======
+
+        Euler
+            Returns the p-th power of the current Euler.
+            Returns the inverse if p = -1.
+
+        Examples
+        ========
+
+        >>> from sympy import Euler, pi
+        >>> euler = Euler(0, 0, pi / 2, 'ZYX')
+        >>> euler.pow(2)
+        Euler(0,  0,  (-pi), ZYX)
+
+        For negative powers, returns the inverse:
+
+        >>> euler.pow(-1)
+        Euler((-pi/2),  0,  0, XYZ)
+
+        """
+        p = sympify(p)
+        if p == -1:
+            return self.inverse()
+
+        q = self.to_quaternion() ** p
+        return Euler.from_quaternion(q, seq=self.seq)
+
+    @property
+    def is_zero_rotation(self):
+        """Returns true if the angles are all zero"""
+        return all(i.is_zero for i in self.angles)
```

Comment:
This should use `fuzzy_and` rather than `all` so that it handles `is_zero = None`.

Reply:
An right, didn't know that, thanks!